docs/source/reference/envs.rst

```diff
@@ -79,20 +79,33 @@ The following figure summarizes how a rollout is executed in torchrl.
 
 In brief, a TensorDict is created by the :meth:`~.EnvBase.reset` method,
 then populated with an action by the policy before being passed to the
-:meth:`~.EnvBase.step` method which writes the observations, done flag and
+:meth:`~.EnvBase.step` method which writes the observations, done flag(s) and
 reward under the ``"next"`` entry. The result of this call is stored for
 delivery and the ``"next"`` entry is gathered by the :func:`~.utils.step_mdp`
 function.
 
 .. note::
 
   The Gym(nasium) API recently shifted to a splitting of the ``"done"`` state
-  into a ``terminated`` (the env is done and results should not be trusted)
-  and ``truncated`` (the maximum number of steps is reached) flags.
-  In TorchRL, ``"done"`` usually refers to ``"terminated"``. Truncation is
-  achieved via the :class:`~.StepCounter` transform class, and the output
-  key will be ``"truncated"`` if not chosen to be something else (e.g.
-  ``StepCounter(max_steps=100, truncated_key="done")``).
+  into a ``termination`` (the env is done and results should not be trusted)
+  and ``truncation`` (the maximum number of steps is reached) flags.
+  In TorchRL, ``"done"`` strictly refers to ``"termination"``.
+  If the environment provides it (eg, Gymnasium), the truncation entry is also
+  updated. In this case, a ``"stop"`` entry can also be provided, which represents
+  the union of ``"done"`` (task completion) and ``"truncated"`` (task interruption).
+  If the environment carries a single value, it will interpreted as a completion
+  (``"done"``) signal by default (and not ``"stop"`` or ``"truncated"``). The
+  reason for this choice is that without any more information, one has to assume
+  task completion for follow-up tasks (eg, computing value estimations), so any
+  unspecified end-of-trajectory signal will eventually need to be interpreted
+  with a worst-case-scenario strategy. Another motivation is that a ``"stop"``
+  signal will be modified by a ``"truncation"`` implemented, for instance, via
+  a :class:`~.StepCounter` transform, and the original ``"stop"`` signal
+  will be lost.
```

Comment:
Not sure to understand this. Could you clarify?

Reply:
I added a little something